cmd/sonarExecuteScan.go

```diff
@@ -0,0 +1,216 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	FileUtils "github.com/SAP/jenkins-library/pkg/piperutils"
+	SliceUtils "github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+type sonarSettings struct {
+	Binary      string
+	Environment []string
+	Options     []string
+}
+
+var sonar sonarSettings
+
+var execLookPath = exec.LookPath
+var fileUtilsExists = FileUtils.FileExists
+var fileUtilsUnzip = FileUtils.Unzip
+var osRename = os.Rename
+
+func sonarExecuteScan(options sonarExecuteScanOptions, telemetryData *telemetry.CustomData) {
+	runner := command.Command{}
+	// reroute command output to loging framework
+	runner.Stdout(log.Entry().Writer())
+	runner.Stderr(log.Entry().Writer())
+
+	client := piperhttp.Client{}
+	client.SetOptions(piperhttp.ClientOptions{Timeout: time.Second * 180})
```

Comment:
I would like to point out https://github.com/SAP/jenkins-library/pull/1257 which probably addresses the root-issue of why such a high timeout is necessary here. I guess without the other change, operations could still be flaky.